pkg/controller/miganalytic/miganalytics_controller_test.go

```diff
@@ -78,3 +88,254 @@ func TestReconcile(t *testing.T) {
+		scheme        *runtime.Scheme
+	}
+	type args struct {
+		request reconcile.Request
+	}
+	tests := []struct {
+		name      string
+		fields    fields
+		args      args
+		want      reconcile.Result
+		wantErr   bool
+		wantReady bool
+	}{
+		{
+			name: "do nothing when Refresh is not set and analytic is Ready",
+			fields: fields{
+				Client: fake.NewFakeClient(&migapi.MigAnalytic{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "miganalytic-00",
+						Namespace: "openshift-migration",
+					},
+					Spec: migapi.MigAnalyticSpec{
+						MigPlanRef: &corev1.ObjectReference{
+							Namespace:  "openshift-migration",
+							Name:       "migplan-00",
+							Kind:       "MigPlan",
+							APIVersion: "migration.openshift.io/v1alpha1",
+						},
+						Refresh: false,
+					},
+					Status: migapi.MigAnalyticStatus{
+						Conditions: migapi.Conditions{
+							List: []migapi.Condition{
+								{
+									Type:     "Ready",
+									Status:   "True",
+									Category: "Required",
+									Message:  "The MigAnalytic is Ready",
+								},
+							},
+						},
+					},
+				}, &testMigPlan, &testMigStorage, &testSourceMigCluster, &testTargetMigCluster, &testSourceNamespace),
+			},
+			want:      reconcile.Result{},
+			wantErr:   false,
+			wantReady: true,
+		},
+		{
+			name: "trigger reconcile when Refresh is set",
+			fields: fields{
+				EventRecorder: record.NewFakeRecorder(20000),
+				Client: fake.NewFakeClient(&migapi.MigAnalytic{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "miganalytic-00",
+						Namespace: "openshift-migration",
+					},
+					Spec: migapi.MigAnalyticSpec{
+						MigPlanRef: &corev1.ObjectReference{
+							Namespace:  "openshift-migration",
+							Name:       "migplan-00",
+							Kind:       "MigPlan",
+							APIVersion: "migration.openshift.io/v1alpha1",
+						},
+						Refresh: true,
+					},
+					Status: migapi.MigAnalyticStatus{
+						Conditions: migapi.Conditions{
+							List: []migapi.Condition{
+								{
+									Type:     "Ready",
+									Status:   "False",
+									Category: "Required",
+									Message:  "The MigAnalytic is Ready",
+								},
+							},
+						},
+					},
+				}, &testMigPlan, &testMigStorage, &testSourceMigCluster, &testTargetMigCluster, &testSourceNamespace),
+			},
+			want:      reconcile.Result{},
+			wantErr:   false,
+			wantReady: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &ReconcileMigAnalytic{
+				Client:        tt.fields.Client,
+				EventRecorder: tt.fields.EventRecorder,
+			}
+			got, err := r.Reconcile(tt.args.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ReconcileMigAnalytic.Reconcile() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ReconcileMigAnalytic.Reconcile() = %v, want %v", got, tt.want)
+			}
+			obj := migapi.MigAnalytic{}
+			if err := r.Client.Get(context.TODO(), types.NamespacedName{
```

Comment:
nit: the obj being looked up can be part of the test's struct `tt.migAnalytics`. It keeps the input and output of the test in the same region and declaritive.